sponsorblock.lua

```diff
@@ -383,17 +398,23 @@ function file_loaded()
     last_skip = {uuid = "", dir = nil}
     chapter_cache = {}
     local video_path = mp.get_property("path")
+    mp.msg.debug("Path: " .. video_path)
+    local video_referer = string.match(mp.get_property("http-header-fields"), "Referer:([^,]+)")
+    mp.msg.debug("Referer: " .. video_referer)
+
     local youtube_id1 = string.match(video_path, "https?://youtu%.be/([%w-_]+).*")
     local youtube_id2 = string.match(video_path, "https?://w?w?w?%.?youtube%.com/v/([%w-_]+).*")
     local youtube_id3 = string.match(video_path, "/watch.*[?&]v=([%w-_]+).*")
     local youtube_id4 = string.match(video_path, "/embed/([%w-_]+).*")
+    local youtube_id5 = string.match(video_referer, "/watch.*[?&]v=([%w-_]+).*")
     local local_pattern = nil
     if options.local_pattern ~= "" then
         local_pattern = string.match(video_path, options.local_pattern)
     end
-    youtube_id = youtube_id1 or youtube_id2 or youtube_id3 or youtube_id4 or local_pattern
+    youtube_id = youtube_id1 or youtube_id2 or youtube_id3 or youtube_id4 or youtube_id5 or local_pattern
     if not youtube_id or string.len(youtube_id) < 11 or (local_pattern and string.len(youtube_id) ~= 11) then return end
     youtube_id = string.sub(youtube_id, 1, 11)
+    mp.msg.info("Found YouTube ID: " .. youtube_id)
```

Comment:
I don't mind having a debug message for this, but I don't want to pollute default logs.